enterprise/server/remote_execution/cgroup/cgroup.go

```diff
@@ -31,6 +32,105 @@ const (
 	cidPlaceholder = "{{.ContainerID}}"
 )
 
+// min/max weight values for cgroup2 weight-based control
+const (
+	minWeight = 1
+	maxWeight = 10_000
+)
+
+// Setup configures the cgroup at the given path with the given settings.
+// IO limits are applied to the given block device, if specified.
+func Setup(ctx context.Context, path string, s *scpb.CgroupSettings, blockDevice *block_io.Device) error {
+	m, err := settingsMap(s, blockDevice)
+	if err != nil {
+		return err
+	}
+	for name, value := range m {
+		settingFilePath := filepath.Join(path, name)
+		if err := os.WriteFile(settingFilePath, []byte(value), 0); err != nil {
+			if os.IsNotExist(err) {
+				log.CtxWarningf(ctx, "Failed to set up cgroup file %q: %s", settingFilePath, err)
+			} else {
+				return fmt.Errorf("write %q: %w", settingFilePath, err)
+			}
+		}
+	}
+	return nil
+}
+
+func settingsMap(s *scpb.CgroupSettings, blockDevice *block_io.Device) (map[string]string, error) {
+	m := map[string]string{}
+	if s.CpuWeight != nil {
+		m["cpu.weight"] = fmt.Sprintf("%d", s.GetCpuWeight())
+	}
+	if s.CpuQuotaPeriodUsec != nil {
+		m["cpu.max"] = fmt.Sprintf("%d %s", s.GetCpuQuotaLimitUsec(), strconv.Itoa(int(s.GetCpuQuotaPeriodUsec())))
```

Comment:
why formatting as int for one field and string for the other?

Also, do you want to clamp these values to their ranges?

Reply:
oops, fixed. I didn't see a min/max for CPU quota in docs?